### ISSUE TITLE
IDE0072: False positive when using switch expression and pattern matching with "or" to default case

### ISSUE THREAD

**Issue by seangwright** [dotnet/roslyn#58468]
**Version Used**: 

```
> dotnet --version
6.0.101
```

**Steps to Reproduce**:

```csharp
enum Greeting
{
    Hello,
    Goodbye
};

Greeting greeting = Greeting.Hello;

string message =>greeting switch
{
    Greeting.Hello => "Hey!",
    Greeting.Goodbye or _ => "Not sure what to say 🤔"
};
```

**Expected Behavior**:

Does not trigger `csharp(IDE0072)` which states I need to `"Populate switch"`.

**Actual Behavior**:

Triggers  `csharp(IDE0072)`.

**Note**

This is a common pattern for me because I don't want skip handling the specific case that has the same result as the default.

If I change the enum and remove `Goodbye`, I should get errors, but this analyzer error wants me to either duplicate the result value (`"Not sure what to say 🤔"`) which means I have to make future changes in 2 places, or remove `Greeting.Goodbye` from the expression and have the default handle it, which removes an intentional reference to a specific value.

This is slightly related to this issue https://github.com/dotnet/roslyn/issues/50982

**Comment by CyrusNajmabadi** [comment]
@seangwright can you give a full example?  The above code is illegal to begin with.  I tried modifying it to make it legal, but i don't repro the problem you mentioned.

Thanks!

**Comment by Youssef1313** [comment]
@CyrusNajmabadi I can repro with the above code (just changed `=>` to `=` so it does compile), and with the following `.editorconfig`:

```ini
[*.cs]

# IDE0072: Add missing cases
dotnet_diagnostic.IDE0072.severity = warning
```

![image](https://user-images.githubusercontent.com/31348972/147739217-08dfe792-bcd0-431b-b02f-7339fc905c34.png)

Two problems I see:

- The reported false positive
    - The false positive makes sense to me per the product code:
        https://github.com/dotnet/roslyn/blob/d9ad24a6d5719a1556f3ee2c16a9103cf36ab192/src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchExpressionHelpers.cs#L72-L73
    `IsDefault` could be renamed to `HasDefault` and its implementation be modified according to the new name. This should have been done as part of https://github.com/dotnet/roslyn/pull/50984, but I haven't thought of this scenario.
- Duplicate diagnostic in IDE (If I enforced code style on build, there is only one diagnostic which is correct - so the duplication seems an IDE-only issue)
